src/apps/explorer/components/TransanctionBatchGraph/index.tsx

```diff
@@ -208,23 +210,32 @@ function TransanctionBatchGraph({
   const [elements, setElements] = useState<ElementDefinition[]>([])
   const cytoscapeRef = useRef<Cytoscape.Core | null>(null)
   const cyPopperRef = useRef<PopperInstance | null>(null)
+  const [resetZoom, setResetZoom] = useState<boolean | null>(null)
   const theme = useTheme()
-  const heightSize = window.innerHeight - HEIGHT_HEADER_FOOTER
+  const { innerHeight } = useWindowSizes()
+  const heightSize = innerHeight && innerHeight - HEIGHT_HEADER_FOOTER
   const setCytoscape = useCallback(
     (ref: Cytoscape.Core) => {
       cytoscapeRef.current = ref
-      cytoscapeRef.current.layout(getLayout()).run()
-      cytoscapeRef.current.fit()
+      const updateLayout = (): void => {
+        ref.layout(getLayout()).run()
+        ref.fit()
+      }
+      ref.on('resize', () => {
+        updateLayout()
+      })
+      updateLayout()
     },
     [cytoscapeRef],
   )
 
   useEffect(() => {
     setElements([])
-    if (error || isLoading || !networkId) return
+    if (error || isLoading || !networkId || !heightSize) return
 
     setElements(getNodes(txSettlement, networkId, heightSize))
-  }, [heightSize, error, isLoading, networkId, txSettlement])
+    setResetZoom(null)
+  }, [error, isLoading, txSettlement, networkId, heightSize, resetZoom])
```

Comment:
I have used the `resetZoom` here as we had agreed. I also added the screen size listener with the solution we discovered for issue #44 .